src/sections/Home/Partners-home/index.js

```diff
@@ -8,27 +8,57 @@ import Slider from "react-slick";
 import "slick-carousel/slick/slick.css";
 import "slick-carousel/slick/slick-theme.css";
 
-const settings = {
-  initialSlide: 1,
-  lazyLoad: true,
-  arrows: false,
-  dots: false,
-  infinite: true,
-  speed: 500,
-  centerMode: true,
-  variableWidth: true,
-  autoplay: true,
-  autoplaySpeed: 1500,
-  className: "partner-slider",
-  responsive: [
-    {
-      breakpoint: 1400,
-      settings: "unslick"
-    }
-  ]
-};
+const PartnerImage = ({ partner }) => (
+  <div className={partner.innerDivStyle}>
+    <img
+      className="partner-image"
+      id={partner.name}
+      src={partner.imageLink}
+      alt={partner.name}
+      width={partner.imageWidth}
+      height={partner.imageHeight}
+    />
+  </div>
+);
 
 const Projects = () => {
+  const [imagesLoaded, setImagesLoaded] = useState(false);
+
+  useEffect(() => {
+    const imagePromises = partners.map(partner => {
+      return new Promise((resolve, reject) => {
+        const img = new Image();
+        img.onload = resolve;
+        img.onerror = reject;
+        img.src = partner.imageLink;
+      });
+    });
+
+    Promise.all(imagePromises)
```

Comment:
why we using promise here? Will any one image anytime fail to load then all images will fail.

Reply:
Promises seem to be ironically named. 😆